api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -126,55 +126,139 @@ data LogRecordArguments body = LogRecordArguments
   , observedTimestamp :: Maybe Timestamp
   , context :: Maybe Context
   , severityText :: Maybe Text
-  , severityNumber :: Maybe Int
+  , severityNumber :: Maybe SeverityNumber
   , body :: body
   , attributes :: H.HashMap Text AnyValue
   }
 
 
-data SeverityNumber = SeverityNumber {shortName :: Maybe Text, severityInt :: !Int64} deriving (Read, Show)
```

Comment:
This data structure is a bit confusing. It is odd that it contains both `severityInt` which is the actual data being represented and also `shortName` which is kind of Metadata about the number. An `Enum` seems less overcomplicated.